liquibase-core/src/main/java/liquibase/command/CommandFactory.java

```diff
@@ -113,7 +113,14 @@ private void consolidateCommandArgumentsForCommand(CommandDefinition commandDefi
             String[][] names = step.defineCommandNames();
             if (names != null) {
                 for (String[] name : names) {
-                    stepArguments.addAll(this.commandArgumentDefinitions.getOrDefault(StringUtil.join(name, " "), new HashSet<>()));
+                    for (CommandArgumentDefinition<?> command :
+                            this.commandArgumentDefinitions.getOrDefault(StringUtil.join(name, " "), new HashSet<>())) {
+                        // uses the most specialized version of the argument, allowing overrides
+                        if (!stepArguments.add(command)) {
```

Comment:
This code confuses me. I'm sure I'm misreading but it looks like it's saying:

IF the command is NOT in the hashset
THEN remove the command from the hashset
AND add the command to the hashset

If it's not there, why do we need to remove it??